adb/fastboot_debug.py

```diff
@@ -12,33 +12,121 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Fastboot debugging binary.
+
+"""Fastboot in python.
 
 Call it similar to how you call android's fastboot. Call it similar to how you
 call android's fastboot, but this only accepts usb paths and no serials.
 """
+
+import argparse
+import inspect
+import logging
 import sys
 
-import gflags
 import progressbar
 
 import common_cli
 import fastboot
 
-gflags.ADOPT_module_key_flags(common_cli)
 
-FLAGS = gflags.FLAGS
+def Devices(args):
+  """Lists the available devices.
+
+  List of devices attached
+  015DB7591102001A        device
+  """
+  for device in fastboot.FastbootCommands.Devices():
+    print('%s\tdevice' % device.serial_number)
+  return 0
+
+
+def InfoCb(message):
+  # Use an unbuffered version of stdout.
+  if not message.message:
+    return
+  sys.stdout.write('%s: %s\n' % (message.header, message.message))
+  sys.stdout.flush()
+
+
+def main():
+  common = common_cli.GetCommonArguments()
+  device = common_cli.GetDeviceArguments()
+  device.add_argument(
+      '--chunk', type=int, default=1024, metavar='1024',
+      help='Size of packets to write to usb. For older devices, it may be '
+           'required to use 4.')
+  parents = [common, device]
+
+  parser = argparse.ArgumentParser(
+      description=sys.modules[__name__].__doc__, parents=[common])
+  subparsers = parser.add_subparsers(title='Commands', dest='command_name')
+
+  subparser = subparsers.add_parser(
+      name='help', help='Prints the commands available')
 
+  subparser = subparsers.add_parser(
+      name='devices', help='Lists the available devices', parents=[common])
+  #subparser.add_argument(
+  #    '--output_port_path', action='store_true',
+  #    help='Outputs the port_path alongside the serial')
 
-def KwargHandler(kwargs, argspec):
+  subparser = common_cli.MakeSubparser(
+      subparsers, parents, fastboot.FastbootCommands.Continue)
 
+  subparser = common_cli.MakeSubparser(
+      subparsers, parents, fastboot.FastbootCommands.Download)
+  subparser.add_argument(
+      'host_path', action=common_cli.PositionalArg,
+      help='Filename on the host to push')
+
+  subparser = common_cli.MakeSubparser(
+      subparsers, parents, fastboot.FastbootCommands.Erase)
```

Comment:
Can you add an argument called 'positional' to MakeSubparser that takes {argument: help}? That will reduce a ton of this boilerplate